app/components/ui/Section/OppEventCardSection.tsx

```diff
@@ -1,86 +1,76 @@
-import imageUrlBuilder from "@sanity/image-url";
-import { SanityImageSource } from "@sanity/image-url/lib/types/types.d";
 import { HomePageSection } from "pages/HomePage/components/Section";
-import { Slug } from "pages/HomePage/components/Section/Section";
 import React from "react";
-import { BackgroundColorVariant } from "models";
-import { client } from "../../../sanity";
+import { StrapiModel } from "models/Strapi";
 import { OppEventCard } from "../Cards/OppEventCard";
 import styles from "./OppEventCardSection.module.scss";
 import { Loader } from "../Loader";
 
-export interface EventData {
-  slug: Slug;
-  image: SanityImageSource;
-
-  /* Event Types*/
-  title?: string;
-  date?: {
-    endDate: Date;
-    startDate: Date;
-    _type: string;
-  };
-
-  /* Opportunity Types*/
-  name?: string;
-  start_date?: Date;
-  end_date?: Date;
-}
-
-export interface OppEventCardData {
-  header: string;
-  subheader?: string;
-  backgroundColor: BackgroundColorVariant;
-  link: {
-    label: string;
-    slug: Slug;
-  };
-  opportunityCards?: EventData[];
-  eventCards?: EventData[];
-}
-
 interface OppEventCardSectionProps {
   sectionType: "event" | "opportunity";
-  sectionData: OppEventCardData;
+  sectionData: StrapiModel.ContentBlock;
+  events?: { attributes: StrapiModel.Event }[];
+  opportunities?: { attributes: StrapiModel.Opportunity }[];
```

Comment:
Thank you for sharing Michael's talk! To keep things simpler, I think moving towards multiple container components, or even just hooks for each data model that transform the Strapi response, is a good way to go. I keep thinking about some of the complexity that has come out of working with the API, and I feel a lot of it is because of that generic useStrapiHook. We may have benefited initially if we went with a hook for each API call needed that returned the model for components to use.